src/Microsoft.AspNetCore.Http/Features/FormOptions.cs

```diff
@@ -12,15 +13,65 @@ public class FormOptions
         public const int DefaultMultipartBoundaryLengthLimit = 128;
         public const long DefaultMultipartBodyLengthLimit = 1024 * 1024 * 128;
 
+        /// <summary>
+        /// Enables full request body buffering. Use this if multiple components need to read the raw stream.
+        /// The default value is false.
+        /// </summary>
         public bool BufferBody { get; set; } = false;
+
+        /// <summary>
+        /// If <see cref="BufferBody"/> is enabled, this many bytes of the body will be buffered in memory.
+        /// If this limit is exceeded then the buffer will be moved to a temp file on disk instead.
+        /// This also applies when buffering individual multipart section bodies.
+        /// </summary>
         public int MemoryBufferThreshold { get; set; } = DefaultMemoryBufferThreshold;
+
+        /// <summary>
+        /// If <see cref="BufferBody"/> is enabled, this is the limit for the total number of bytes that will
+        /// be buffered. Forms that exceed this limit will throw an <see cref="InvalidDataException"/> when parsed.
```

Comment:
curious regarding usage of `MemoryBufferThreshold` vs. `BufferBodyLengthLimit`: If I want to buffer the data and also be safe from larger payloads, I could set the MemoryBufferThreshold alone...so wondering the use of BufferBodyLengthLimit